content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+
+1. **Open the `DESCRIPTION` file** in your project root.
+
+
+2. **Edit the file** with your own information. Not sure what to write for each field? See [this resource](https://r-pkgs.org/description.html#pkg-description) for suggestions.
+
+3. Finally, make sure to __*Build* > *Install and Restart*__ your package again to include these changes.
+
+
+
+### Getting your tutorial to your audience
+
+Your tutorial package has (finally!) been built, now it's time to get it out there!
+
+1. **[Commit](https://happygitwithr.com/rstudio-git-github.html#make-local-changes-save-commit) and [push](https://happygitwithr.com/rstudio-git-github.html#push-your-local-changes-online-to-github)** to GitHub to share your tutorial package with the world.
+
+1. **Have users *install* your package** from GitHub with `remotes::install_github(“<your-repo>/<package-name>")` (They will also need to install the learnr package.)
+
+    ```{r, eval = FALSE}
+    remotes::install_github("dcossyleon/tutorialpackage")
+    ```
+
+1. **Have users *run* your tutorial** with `learnr::run_tutorial("<tutorial-name>", "<package-name>")`
+
+    ```{r, eval = FALSE}
+    learnr::run_tutorial("lesson1", "tutorialpackage")
+    ```
+
+We recommend including these latter two instructions in a README file, which we introduce below.
+
+
+
+### Create a README file
+
+You can include additional information in a `README.Rmd` file for your package. At a minimum, we recommend you include a line about how users can install and load the tutorials.
+
+1. **Run** `use_readme_rmd()` to create and open a README file.
+
+    ```{r, eval =  FALSE}
+    use_readme_rmd()
+    ```
+
+1. **Edit** the file to meet your needs. Need ideas of what to include? Check out [this resource](https://r-pkgs.org/whole-game.html#use_readme_rmd).
+
+1. **Click Knit** so that it creates a corresponding `README.md` file.
+
+    Or, alternatively, run:
+
+    ```{r, eval = FALSE}
+    usethis::build_readme()
+    ```
+
+1. **Commit and push** your changes to GitHub.
+
+
+
+### Updating your package
+
+After your package has been installed by your users, you can continue to update or add tutorials. Each time you make changes to the contents of your package, you must *Build* then *Install and Restart* the package locally before pushing your changes to GitHub.
+
+Users can then access the updated content by simply re-installing your package.
+
+
+## That's a wrap!
+
+Congratulations, you've created a tutorial package!
+
+From here, there are many ways to extend what you've done. For example, you could decide to bundle up custom data sets in your package, as explained in the [R Packages](https://r-pkgs.org/data.html#data-sysdata) book. And you might consider adding [package documentation](https://r-pkgs.org/man.html#man-packages).
+
+If you'd like to see an example of a tutorial package "in the wild", check out the [GitHub repository for the `learntidymodels` package](https://github.com/tidymodels/learntidymodels).
```

Comment:
If you'd like one more example, this one also has it: https://github.com/rstudio-education/dsbox